src/D2L.Bmx/AwsCredsCreator.cs

```diff
@@ -83,12 +74,26 @@ bool cache
 		AwsRole[] rolesData = HtmlXmlHelper.GetRolesFromSamlResponse( samlResponse );
 
 		if( string.IsNullOrEmpty( role ) ) {
-			if( !nonInteractive ) {
-				string[] roles = rolesData.Select( r => r.RoleName ).ToArray();
-				role = consolePrompter.PromptRole( roles );
-			} else {
+			if( nonInteractive ) {
 				throw new BmxException( "Role value was not provided" );
 			}
+			string[] roles = rolesData.Select( r => r.RoleName ).ToArray();
+			role = consolePrompter.PromptRole( roles );
+		}
+
+		// try getting from cache again even if calling Okta is inevitable (we still avoid the AWS call)
+		if( cache ) {
```

Comment:
Real change part 2 of 2.

Reply:
Seems fine, I think  the code as originally written would have guarded against valid credentials in BMX that Okta no longer believes are valid (user unassigned from app etc...?)

That doesn't seem like something BMX would be able to care for, it could only be done authoritatively on the AWS side by revoking old sessions altogether.

Reply:
> I think the code as originally written would have guarded against valid credentials in BMX that Okta no longer believes are valid (user unassigned from app etc...?)

No, there's no material difference here.

If user is unassigned from the app, and
* if they provide everything upfront, we don't call Okta in both worlds, and they can use the creds just fine
* if they don't provide everything upfront
  * in the old world BMX fails directly
  * in the new world they won't see the account & role they want in the selection prompts